librosa/core/pitch.py

```diff
@@ -616,25 +625,25 @@ def yin(
 
 
 def pyin(
-    y,
+    y: np.ndarray,
     *,
-    fmin,
-    fmax,
-    sr=22050,
-    frame_length=2048,
-    win_length=None,
-    hop_length=None,
-    n_thresholds=100,
-    beta_parameters=(2, 18),
-    boltzmann_parameter=2,
-    resolution=0.1,
-    max_transition_rate=35.92,
-    switch_prob=0.01,
-    no_trough_prob=0.01,
-    fill_na=np.nan,
-    center=True,
-    pad_mode="constant",
-):
+    fmin: float,
+    fmax: float,
+    sr: float = 22050,
+    frame_length: int = 2048,
+    win_length: Optional[int] = None,
+    hop_length: Optional[int] = None,
+    n_thresholds: int = 100,
+    beta_parameters: Tuple[float, float] = (2, 18),
+    boltzmann_parameter: float = 2,
+    resolution: float = 0.1,
+    max_transition_rate: float = 35.92,
+    switch_prob: float = 0.01,
+    no_trough_prob: float = 0.01,
+    fill_na: Optional[float] = np.nan,
+    center: bool = True,
+    pad_mode: Union[str, Callable] = "constant",
```

Comment:
i think callable here might be a typo/incorrect documentation

Reply:
Seems legit, np.pad does accept a function

Reply:
Ah I see that now in the numpy docs, but it had not made it into the librosa docs.  We probably should do a quick sweep through the package for uses of padding just to make sure we're consistent throughout.

Reply:
Done